docs/data/data-grid/rows/rows.md

```diff
@@ -95,9 +95,42 @@ If you need some rows to have different row heights this can be achieved using t
 > The grid bases on the referential value of these props to cache their values and optimize the rendering.
 >
 > ```tsx
-> const handleGetRowHeight = React.useCallback(() => { ... }, []);
+> const getRowHeight = React.useCallback(() => { ... }, []);
 >
-> <DataGridPro getRowHeight={handleGetRowHeight} />
+> <DataGridPro getRowHeight={getRowHeight} />
+> ```
+
+### Dynamic row height
+
+Instead of a fixed row height, you can let the grid to calculate the height of each row based on its content.
+To do so, return `"auto`" on the function passed to the `getRowHeight` prop.
+
+```tsx
+<DataGrid getRowHeight={() => 'auto'} />
+```
+
+The following demo demonstrantes this option in action:
+
+{{"demo": "DynamicRowHeightGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+The dynamic row height implementaion is based on a lazy approach, which means that the rows are measured as they are rendered.
+Because of this, you may see the size of the scrollbar thumb changing during scroll.
+This side effect happens because a row height estimation is used while a row is not rendered, then this value is replaced once the true measurement is obtained.
+You can configure the estimated value used by passing a function to the `getEstimatedRowHeight` prop.
+If not provided, the default row height of `52px` is used as estimation.
+It's recommended to pass this prop if the content deviates too much from the default value.
+
+```tsx
+<DataGrid getRowHeight={() => 'auto'} getEstimatedRowHeight={() => 200} />
+```
+
+{{"demo": "ExpandableCells.js", "bg": "inline", "defaultCodeOpen": false}}
+
+> ⚠ When the height of a row is set to `"auto"`, the final height will follow exactly the content size and ignore the density.
```

Comment:
I'm curious why not support the density. Is it not feasible?

It could be very annoying to users if they need to still support density on their own.

Reply:
If the content has 100px as height, then the row height should be also 100px. With #4858 will be able to vary the padding according to the density.

Reply:
We had the same case for the variable row height. If the user provides their own row height then they are "taking control" over the display and should handle the density change themselves. Technically the `getRowHeight` callback will be called for each row when the density changes so the developer can react to it and maybe return a different value.

Reply:
Now with #4858 merged I updated the demo to react to the selected density.